T012_M1_load_data.py

```diff
@@ -1,55 +1,122 @@
-Ensure that the function calls are added to the end of the main script.
-
-1. Follow the FDR to implement a function called add_average. This function must be placed inside the file Txxx_M1_load_data.py below load_data.
-Function Description:
-● The function will add the average of the student’s grades (G1, G2 and G3) as an additional attribute to the dictionary.
-● It takes one input parameter (regardless of the key): (1) a dictionary
-● It returns the dictionary updated with the average grade. Sample output (assuming school as key):
-{ 'GP' : [ {'Age': 18, 'StudyTime': 1.8, 'Failures': 0, 'Health': 3, 'Absences': 6, 'G1': 5, 'G2': 6, 'G3': 6, 'G_Avg': 5.67 },
-{another element},
-… ],
-'MB' : [ {'Age': 16, 'StudyTime': 2.6, 'Failures': 0, 'Health': 3, 'Absences': 12, 'G1': 10, 'G2': 12, 'G3': 12, 'G_Avg': 11.33 },
-{another element},
-… ],
-… }
-2. Test your function. For this lab, you can just call your function to ensure that the data is loaded correctly. For testing purposes, use student-mat.csv. 
-Ensure that the function calls are added to the end of the main script.
-
-1. Submit your module (Txxx_M1_load_data.py) on Brightspace (One submission per team). Before you submit, ensure:
-  a. All the active team members’ names are at the top of the file as a Python comment. If a name is missing, that person will receive zero. We will assume that 
-     the person did not participate in the task.
-  b. The functions have the proper names.
-  c. You have included type annotations and docstrings. Do not forget to include the preconditions, if any.
-  d. At the end of the file, you have the main script in which the functions are called, and all testing scenarios are covered. For this lab, function calls are 
-     enough to ensure that the data is properly loaded.
-  e. The file name is correct.
-  f. Run Txxx_M1_load_data.py one more time to ensure that there are no errors!
-"""
 
+def student_school_dictionary(filepath: str) -> dict[list]:
+    """
+    Return a dictionary with loaded data given a filepath of a .csv file.
+    Precondition: filepath is a correct and existing file path to a .csv file
+    >>>student_school_dictionary("student-mat.csv")
+    {'GP': [{'Age': 18, 'StudyTime': 2, 'Failure': 0, 'Health': 3, 'Absences': 6, 'G1': 5, 'G2': 6, 'G3': 6},
+            {'Age': 17, 'StudyTime': 2, 'Failure': 0, 'Health': 3, 'Absences': 4, 'G1': 5, 'G2': 5, 'G3': 6},
+            ...],
+     'MB': [{'Age': 18, 'StudyTime': 2, 'Failure': 0, 'Health': 3, 'Absences': 6, 'G1': 5, 'G2': 6, 'G3': 6},
+            {'Age': 17, 'StudyTime': 2, 'Failure': 0, 'Health': 3, 'Absences': 4, 'G1': 5, 'G2': 5, 'G3': 6}
+            ...],
+      ...}
+    """
+    KEY_TYPE = 'School'
+    file = open(filepath, "r")
+    master_dictionary = {}
+    first_run = True
+
+    for line in file:
+        temp_dict = {}
```

Comment:
```suggestion
```
Esteban, I have suggested the change to my PR to reflect Milan's PR above mine in the PR queue. We have 2 options: can either just roll his change in here (I'll accept my own code suggestion if you can't) OR:
option 2: just merge Milan's PR before you merge mine (because that way the conflict it generates (which we'll need to resolve) will only be a 1-line conflict with my PR -- whereas if you merge my PR first it's going to be a nightmare resolving conflicts on his PR -- we'll wind up just wanting to close his and ask him to re-file))